.github/workflows/pre-integration.yml

```diff
@@ -21,11 +21,12 @@ jobs:
     name: Test JDK ${{ matrix.java }}, ${{ matrix.os }}, ${{ matrix.profile }}
 
     steps:
-    - uses: actions/checkout@v2
+    - uses: actions/checkout@v3
     - name: Set up JDK ${{ matrix.java }}
-      uses: actions/setup-java@v1
+      uses: actions/setup-java@v3
```

Comment:
> What is the benefit of switching to v3 at this time? -- @shipilev 

It runs on newer version of infrastructure (e.g. Node 12 -> 16), which might have performance improvements, vulnerability fixes etc.

I would ask: what's the reason to keep using an unmaintained almost 1 year old version? (really curious)

Reply:
Well, mostly "if it is not broken, don't fix it", but yes, we should switch to actively maintained version.

Reply:
@shipilev funnily enough: it's "broken" now ...

See warnings on https://github.com/shipilev/jmh/actions/runs/3243894873

https://github.blog/changelog/2022-09-22-github-actions-all-actions-will-begin-running-on-node16-instead-of-node12/

Reply:
I'll handle it in #84.

Reply:
Yep 👍🏻, just wanted to follow the conversation here :)